src/charm.py

```diff
@@ -144,6 +312,7 @@ def _generate_config_file(self) -> bool:
             nrf_ip=_get_pod_ip(),  # type: ignore[arg-type]
             database_name=DATABASE_NAME,
             nrf_sbi_port=NRF_SBI_PORT,
+            scheme="https" if self._certificate_is_stored() else "http",
```

Comment:
All the code for certificate will only work for the leader, and I see that the `nrf` integration is only handled by the leader.

What happens if this gets scaled up and the leader changes?

I think we might voluntarily send the charm in error if it is not leader inside `__init__` if we do not support scaling at this point. It would at least communicate to the user that their setup is broken.

Reply:
Right now, nothing "scaling" related has been validated and I suspect scaling simply won't work at all. Most NF's aren't built to scale as they are.

Reply:
Done